api/api.go

```diff
@@ -502,3 +550,102 @@ func OrbSource(ctx context.Context, logger *logger.Logger, namespace string, orb
 
 	return response.Orb.Versions[0].Source, nil
 }
+
+// ListNamespaceOrbs queries the API to find all orbs belonging to the given
+// namespace. Prints the orbs and their jobs and commands to the supplied
+// logger.
+func ListNamespaceOrbs(ctx context.Context, logger *logger.Logger, namespace string) error {
+	// Define a structure that matches the result of the GQL
+	// query, so that we can use mapstructure to convert from
+	// nested maps to a strongly typed struct.
+	type namespaceOrbResponse struct {
+		RegistryNamespace struct {
+			Name string
+			Orbs struct {
+				Edges []struct {
+					Cursor string
+					Node   struct {
+						Name     string
+						Versions []struct {
+							Version string
+							Source  string
+						}
+					}
+				}
+				TotalCount int
+				PageInfo   struct {
+					HasNextPage bool
+				}
+			}
+		}
+	}
+
+	request := graphql.NewRequest(`
+query namespaceOrbs ($namespace: String, $after: String!) {
+	registryNamespace(name: $namespace) {
+		name
+		orbs(first: 20, after: $after) {
+			edges {
+				cursor
+				node {
+					versions {
+						source
+						version
+					}
+					name
+				}
+			}
+			totalCount
+			pageInfo {
+				hasNextPage
+			}
+		}
+	}
+}
+`)
+
+	address, err := GraphQLServerAddress(EnvEndpointHost())
+	if err != nil {
+		return err
+	}
+	client := client.NewClient(address, logger)
+
+	var result namespaceOrbResponse
+	currentCursor := ""
+
+	for {
+		request.Var("after", currentCursor)
+		request.Var("namespace", namespace)
+		err := client.Run(ctx, request, &result)
+
+		if err != nil {
+			return errors.Wrap(err, "GraphQL query failed")
+		}
+
+	NamespaceOrbs:
+		for i := range result.RegistryNamespace.Orbs.Edges {
+			edge := result.RegistryNamespace.Orbs.Edges[i]
+			currentCursor = edge.Cursor
+			if len(edge.Node.Versions) > 0 {
+				v := edge.Node.Versions[0]
+
+				// Print the orb name and first version returned by the API
+				logger.Infof("%s (%s)", edge.Node.Name, v.Version)
+
+				// Parse the orb source to print its commands, executors and jobs
+				var o Orb
+				err := yaml.Unmarshal([]byte(edge.Node.Versions[0].Source), &o)
+				if err != nil {
+					logger.Error(fmt.Sprintf("Corrupt Orb %s %s", edge.Node.Name, v.Version), err)
+					continue NamespaceOrbs
+				}
+				logger.Info(o.String())
```

Comment:
This function should probably return the `[]Orb` so that the `cmd` package can print them.

The `Orb` struct could have a `Name` and `Version` field as well.